code/obj/item.dm

```diff
@@ -538,6 +548,10 @@
 /obj/item/temperature_expose(datum/gas_mixture/air, temperature, volume)
 	if (src.burn_possible && !src.burning)
 		if ((temperature > T0C + src.burn_point) && prob(5))
+			for (var/obj/item/I in get_turf(src))
+				if (I.firesource)
+					src.combust(I)
+					break
```

Comment:
shouldn't call combust here as we are also calling right below. Instead let's just find the fire source and pass it down, or `null` if we can't find one

Note that github is annoying and won't let me select the line just below your changes, but it should be removed as I'm including it in here with the change of adding the `firesource` var as one of its args

```suggestion
			var/obj/item/firesource = null
			for (var/obj/item/I in get_turf(src))
				if (I.firesource)
					firesource = I
					break
			src.combust(firesource)
```